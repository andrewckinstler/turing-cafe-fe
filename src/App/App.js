import React, { Component } from 'react';
import { Reservations } from '../Reservations/Reservations'
import { Form } from '../Form/Form';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: [],
      loading: true
    }
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(res => res.json())
      .then(data => this.setState({ reservations: data, loading: false }))
  }

  addReservation = reservation => {
    this.setState({ reservations: [...this.state.reservations, reservation]})
    this.postReservation(reservation)
  }

  postReservation = reservation => {
    const url = 'http://localhost:3001/api/v1/reservations'
    const options = {
      method: 'POST',
      body: JSON.stringify(reservation),
      headers: {
        'Content-Type': 'application/json'  
      }
    }
    
    return fetch(url, options)
            .then(res => {
              if(!res.ok) {
                throw Error('Error creating reservation')
              }
              return res.json()})
  }



  render() {
    const { reservations, loading } = this.state;
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form addReservation={this.addReservation} />
        </div>
        {!loading
        ? <div className='resy-container'>
        <Reservations props={reservations} />
      </div>
      :<h2>Loading...</h2>}
        
      </div>
    )
  }
}

export default App;
