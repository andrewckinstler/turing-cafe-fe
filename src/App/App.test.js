import React from 'react';
import { shallow } from 'enzyme'
import App from './App';

it('should match a snapshot', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
});

it('should update state when addReservation is called', () => {
  const wrapper = shallow(<App />)
  const mockReservation = {
    name: 'Sharon',
    date: '01/01',
    time: '8:00 pm',
    number: 6
  }
  const expected = [mockReservation]

  expect(wrapper.state('reservations')).toEqual([])
  wrapper.instance().addReservation(mockReservation)
  expect(wrapper.state('reservations')).toEqual(expected)
});
