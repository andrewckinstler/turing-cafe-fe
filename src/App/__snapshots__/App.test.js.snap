// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      addReservation={[Function]}
    />
  </div>
  <h2>
    Loading...
  </h2>
</div>
`;
